foia_hub/templates/contacts/index.html

```diff
@@ -13,26 +13,34 @@
     {% endif %}
     {% if not query %}
       <h2><em>Federal Agencies</em></h2>
-    {% endif %}    
-  
-    <ul>
-      {% for agency in agencies %}
-          <li>
-            <div class="agencies--name">
-              <h3><a href="/contacts/{{agency.slug }}">{{ agency.name }}</a></h3>
-            </div>
-            <div class="agencies--description">
-              <p>
-                {% if agency.description %}
-                  {{ agency.description }}
-                {% else %}
-                  Click to find out more about this agency.
-                {% endif %}
-              </p>
-            </div>
-          </li>
+    {% endif %}
+      <h3>Results: {{agencies|length}}</h3>
+      <p>
+          {% for letter in agencies|groupby('name.0') %}
```

Comment:
You could set the results as a variable, e.g. `{% set grouped_agencies = agencies|groupby('name.0') %}` to make sure the logic is consistent, but you'd still be iterating twice. That seems okay to me. If the first loop ever gets too complicated (and you want to keep both loops visible in the same screen), consider a [macro](http://jinja.pocoo.org/docs/dev/templates/#macros).